tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/caches/ModelValueCache.scala

```diff
@@ -22,15 +22,43 @@ class ModelValueCache(solverContext: SolverContext)
     val (utype, _) = typeAndIndex
     val newArena = arena.appendCell(ConstT(utype))
     val newCell = newArena.topCell
-    // the freshly created cell should differ from the others
-    for (
-        other <- values().withFilter { v =>
-          v.cellType == ConstT(utype) // compare just with same sorted cells
-        }
-    ) {
-      solverContext.assertGroundExpr(OperEx(ApalacheOper.distinct, newCell.toNameEx, other.toNameEx))
-    }
+    // The fresh cell should differ from the previously created cells.
+    // We use the SMT constraint (distinct ...).
+    val others = values().filter(_.cellType == ConstT(utype)).map(_.toNameEx).toSeq
+    solverContext.assertGroundExpr(OperEx(ApalacheOper.distinct, newCell.toNameEx +: others: _*))
     solverContext.log("; cached \"%s\" to %s".format(typeAndIndex, newCell))
     (newArena, newCell)
   }
+
+  /**
+   * Cache multiple values at once. This method is more efficient than caching multiple values with [[create]].
+   *
+   * @param arena
+   *   the arena to start with
+   * @param utype
+   *   the value type
```

Comment:
value type of what?

Reply:
Fixed: the name of the uninterpreted type to use